scripts/updatedb.js

```diff
@@ -0,0 +1,638 @@
+    text = tryFixingLine(text)
+    if (!reValid.test(text)) { return null }
+  }
+  const a = [] // Initialize array to receive values.
+  text.replace(reValue, // "Walk" the string using replace with callback.
+    function (m0, m1, m2, m3) {
+      // Remove backslash from \' in single quoted values.
+      if (m1 !== undefined) a.push(m1.replace(/\\'/g, "'"))
+      // Remove backslash from \" in double quoted values.
+      else if (m2 !== undefined) a.push(m2.replace(/\\"/g, '"').replace(/\\'/g, "'"))
+      else if (m3 !== undefined) a.push(m3)
+      return '' // Return empty string.
+    })
+  // Handle special case of empty last value.
+  if (/,\s*$/.test(text)) a.push('')
+  return a
+}
+
+function getHTTPOptions (downloadUrl) {
+  const options = new URL(downloadUrl)
+  options.headers = {
+    'User-Agent': userAgent
+  }
+
+  if (process.env.http_proxy || process.env.https_proxy) {
+    try {
+      const HttpsProxyAgent = require('https-proxy-agent')
+      options.agent = new HttpsProxyAgent(process.env.http_proxy || process.env.https_proxy)
+    } catch (e) {
+      console.error('Install https-proxy-agent to use an HTTP/HTTPS proxy')
+      process.exit(-1)
+    }
+  }
+
+  return options
+}
+
+function check (database, cb) {
+  if (args.indexOf('force') !== -1) {
+    // we are forcing database upgrade,
+    // so not even using checksums
+    return cb(null, database)
+  }
+
+  const checksumUrl = database.checksum
+
+  if (typeof checksumUrl === 'undefined') {
+    // no checksum url to check, skipping
+    return cb(null, database)
+  }
+
+  // read existing checksum file
+  fs.readFile(path.join(dataPath, database.type + '.checksum'), function (err, data) {
+    if (!err && data && data.length) {
+      database.checkValue = data
+    }
+
+    console.log('Checking ', database.fileName)
+
+    function onResponse (response) {
+      const status = response.statusCode
+
+      if (status !== 200) {
+        console.log('ERROR'.red + ': HTTP Request Failed [%d %s]', status, http.STATUS_CODES[status])
+        client.abort()
+        process.exit()
+      }
+
+      let str = ''
+      response.on('data', function (chunk) {
+        str += chunk
+      })
+
+      response.on('end', function () {
+        if (str && str.length) {
+          if (str === database.checkValue) {
+            console.log(('Database "' + database.type + '" is up to date').green)
+            database.skip = true
+          } else {
+            console.log(('Database ' + database.type + ' has new data').green)
+            database.checkValue = str
+          }
+        } else {
+          console.log('ERROR'.red + ': Could not retrieve checksum for', database.type, 'Aborting'.red)
+          console.log('Run with "force" to update without checksum')
+          client.abort()
+          process.exit()
+        }
+        cb(null, database)
+      })
+    }
+
+    const client = https.get(getHTTPOptions(checksumUrl), onResponse)
```

Comment:
here I think we can use https://github.com/bitfinexcom/bfx-facs-http instead as it supports both async and cb support + http agents